### PR TITLE
[Security] Add support for dynamic CSRF id with Expression in `#[IsCsrfTokenValid]`

### LINKED ISSUE #19753: [Security] Add support for dynamic CSRF id with Expression in `#[IsCsrf…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#54443
| PR author(s) | @yguedidi
| Merged in    | 7.1

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.1 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alamirault**
Fix https://github.com/symfony/symfony-docs/issues/19753

### CODE REVIEW

**Comment by rcsofttech85** [comment]
you may consider changing your code to this...

```php
    #[IsCsrfTokenValid(new Expression('"delete-item-" ~ args["post"].id'), tokenKey: 'token')]
    public function delete(Post $post): Response
    {
        // ... do something, like deleting an object
    }
```
post argument is mandatory in this case

**Comment by alamirault** [comment]
You're right, updated

**Review by javiereguiluz — APPROVED** [review]

File: security/csrf.rst [hunk 1]
```diff
@@ -181,6 +181,32 @@ attribute on the controller action::
...
+
+    <form action="{{ url('admin_post_delete', { id: post.id }) }}" method="post">
+        {# the argument of csrf_token() is a dynamic id string used to generate the token #}
+        <input type="hidden" name="token" value="{{ csrf_token('delete-item-' ~ post.id) }}">
+
+        <button type="submit">Delete item</button>
+    </form>
+
+The :class:`Symfony\\Component\\Security\\Http\\Attribute\\IsCsrfTokenValid`
+attribute also accept an :class:`Symfony\\Component\\ExpressionLanguage\\Expression`
```
Comment (javiereguiluz) [inline security/csrf.rst]: `also accept` -> `also accepts`

**Comment by OskarStark** [comment]
Thank you Antoine.

**Comment by yguedidi** [comment]
Thank you @alamirault

**Pull request merged**